### PR TITLE
byteman 2.2.1

### PR DESCRIPTION
**Pull request by adriancole**
JVM agent that allows you to do dynamic tracing and fault injection.

cc @adinn

### CODE REVIEW

File: Library/Formula/byteman.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  def install
+    rm_f Dir["bin/*.bat"]
+    libexec.install Dir["*"]
+  end
+
+  def caveats; <<-EOS.undent
+    The home of Byteman is:
+      #{opt_libexec}
+    You may want to add the following to your .bash_profile:
+      export BYTEMAN_HOME=#{opt_libexec}
```
Comment (mikemcquaid) [inline Library/Formula/byteman.rb]: We should write a wrapper script instead.

File: Library/Formula/byteman.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+      export PATH=${PATH}:${BYTEMAN_HOME}/bin
+    EOS
+  end
+
+  test do
+    system "#{opt_libexec}/bin/bmcheck.sh | grep usage"
```
Comment (mikemcquaid) [inline Library/Formula/byteman.rb]: Don't use `grep` here; you can use `shell_output` and `assert_match` instead or just rely on the exit code.

Comment (adriancole) [inline Library/Formula/byteman.rb]: Will do

File: Library/Formula/byteman.rb (hunk 1 above)
Comment (adriancole) [inline Library/Formula/byteman.rb]: Example? Otherwise, if instructions like this are problematic I can remove
them.

Comment (adriancole) [inline Library/Formula/byteman.rb]: removed this section as it wasn't pulling weight anyway.

File: Library/Formula/byteman.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
+class Byteman < Formula
+  homepage "http://byteman.jboss.org/"
+  url "http://downloads.jboss.org/byteman/2.2.1/byteman-download-2.2.1-bin.zip"
+  sha256 "c4f8b5a89163d2ff42f90f6a90025ebd165d1e9046390f97bb5657e4fd1b85bb"
+
+  def install
+    rm_f Dir["bin/*.bat"]
+    libexec.install Dir["*"]
```
Comment (mikemcquaid) [inline Library/Formula/byteman.rb]: Feels a bit weird that there will be nothing installed into `bin`?

Comment (adriancole) [inline Library/Formula/byteman.rb]: Oh. Good point. I basically stopped when typing bmsubmit.sh worked :) will
swing the axe again wrt bin

**Comment by adriancole** [comment]
ps I will bump this to the just released 2.2.2

**Comment by mikemcquaid** [comment]
@adriancole :+1:

**Comment by bfontaine** [comment]
Any update on this?

**Comment by DomT4** [comment]
Ping. Will be closed soon without a response.

**Comment by adriancole** [comment]
sorry to distract. you can close it. I cannot refactor it at the moment and
can pick back up later if it becomes foreground again.

**Pull request closed without merging**